components/layout/display_list/builder.rs

```diff
@@ -2535,12 +2535,25 @@ impl BlockFlow {
             return;
         }
 
-        let sensitivity = if StyleOverflow::Hidden == self.fragment.style.get_box().overflow_x &&
-            StyleOverflow::Hidden == self.fragment.style.get_box().overflow_y
-        {
-            ScrollSensitivity::Script
-        } else {
-            ScrollSensitivity::ScriptAndInputEvents
+        let sensitivity_x = match self.fragment.style.get_box().overflow_x {
+            StyleOverflow::Hidden => ScrollSensitivity::Script,
+            //TODO
+            //StyleOverflow::Clip => ScrollSensitivity::None,
```

Comment:
We should never have `overflow: clip` here, as it would have been rejected at L2522. So we should have
```rust
ComputedOverflow::Clip => unreachable!("A scroll container should not be established with overflow: clip"),
```
Ditto below